ocaml/xapi/repository.ml

```diff
@@ -128,7 +130,8 @@ let sync ~__context ~self ~token ~token_id =
     write_initial_yum_config () ;
     clean_yum_cache repo_name ;
     (* Remove imported YUM repository GPG key *)
-    Xapi_stdext_unix.Unixext.rm_rec (get_repo_config repo_name "gpgdir") ;
+    if Pkg_mgr.(active () = Yum) then
```

Comment:
Please add the value to `S` instead:
```suggestion
    if Pkgs.manager = Yum then
```

This also allows `active` and `mgr` to be dropped  from the interface